src/Field/Definition/ChoiceField.php

```diff
@@ -93,6 +130,57 @@ public function validateData (ComponentContext $context, array $contentPath, mix
 		}
 	}
 
+
+	private function validateMultiSelect (ComponentContext $context, array $contentPath, mixed $data) : void
+	{
+		$context->ensureDataIsValid(
+			$contentPath,
+			$this,
+			$data,
+			[
+				new Type("array"),
+				new All([
+					new NotNull(),
+					new AtLeastOneOf([
+						new Type("string"),
+						new Type("int"),
+					]),
+				]),
+			],
+		);
+
+		\assert(\is_array($data));
+
+		if ($this->required || null !== $this->minimumNumberOfOptions || null !== $this->maximumNumberOfOptions)
+		{
+			$context->ensureDataIsValid(
+				$contentPath,
+				$this,
+				$data,
+				[
+					new Count(
+						min: $this->minimumNumberOfOptions ?? ($this->required ? 1 : null),
+						max: $this->maximumNumberOfOptions,
+						minMessage: "At least {{ limit }} option(s) must be selected.",
+						maxMessage: "You cannot specify more than {{ limit }} options.",
+					),
+				],
+			);
+		}
+
+		$choicesConstraints = $this->choices->getValidationConstraints(true);
```

Comment:
Now that I see that: I really dislike this API. We need to change it in the next major version 😅